pkg/psmdb/container.go

```diff
@@ -181,24 +181,32 @@ func containerArgs(ctx context.Context, cr *api.PerconaServerMongoDB, replset *a
 		"--replSet=" + replset.Name,
 		"--storageEngine=" + string(replset.Storage.Engine),
 		"--relaxPermChecks",
-		"--sslAllowInvalidCertificates",
 	}
 
 	name, err := replset.CustomReplsetName()
 	if err == nil {
 		args[4] = "--replSet=" + name
 	}
 
-	if cr.Spec.UnsafeConf {
+	if cr.TLSEnabled() {
+		args = append(args, "--sslAllowInvalidCertificates")
+		if cr.Spec.TLS.Mode == api.TLSModeAllow {
+			args = append(args,
+				"--clusterAuthMode=keyFile",
+				"--keyFile="+mongodSecretsDir+"/mongodb-key",
+			)
+		} else {
+			args = append(args, "--clusterAuthMode=x509")
+		}
+	} else if (cr.CompareVersion("1.16.0") >= 0 && cr.Spec.Unsafe.TLS) || (cr.CompareVersion("1.16.0") < 0 && cr.Spec.UnsafeConf) {
```

Comment:
What do you think about creating a `cr.UnsafeTLSDisabled()` method to replace this long condition check?

Reply:
https://github.com/percona/percona-server-mongodb-operator/pull/1504/commits/6badccfc800b3b7cd469a4a1bdecc9223eb4ab54